Artifacts/MsSql/Structure/Security/0040-Tables - RLPhase2.sql

```diff
@@ -0,0 +1,132 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+---------------------------- [dbo].[ClaimSetResourceClaimActionAuthorizations] --------------------------------------------------------
+
+CREATE TABLE [dbo].[ClaimSetResourceClaimActionAuthorizations](
+	[ClaimSetResourceClaimActionAuthorizationId] [int] IDENTITY(1,1) NOT NULL,
+	[ValidationRuleSetNameOverride] [nvarchar](255) NULL,
+	[Action_ActionId] [int] NOT NULL,
+	[ClaimSet_ClaimSetId] [int] NOT NULL,
+	[ResourceClaim_ResourceClaimId] [int] NOT NULL,
+ CONSTRAINT [PK_dbo.ClaimSetResourceClaimActionAuthorizations] PRIMARY KEY CLUSTERED 
+(
+	[ClaimSetResourceClaimActionAuthorizationId] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
+) ON [PRIMARY]
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizations]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizations_dbo.Actions_Action_ActionId] FOREIGN KEY([Action_ActionId])
+REFERENCES [dbo].[Actions] ([ActionId])
+ON DELETE CASCADE
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizations] CHECK CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizations_dbo.Actions_Action_ActionId]
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizations]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizations_dbo.ClaimSets_ClaimSet_ClaimSetId] FOREIGN KEY([ClaimSet_ClaimSetId])
+REFERENCES [dbo].[ClaimSets] ([ClaimSetId])
+ON DELETE CASCADE
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizations] CHECK CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizations_dbo.ClaimSets_ClaimSet_ClaimSetId]
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizations]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizations_dbo.ResourceClaims_ResourceClaim_ResourceClaimId] FOREIGN KEY([ResourceClaim_ResourceClaimId])
+REFERENCES [dbo].[ResourceClaims] ([ResourceClaimId])
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizations] CHECK CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizations_dbo.ResourceClaims_ResourceClaim_ResourceClaimId]
+GO
+
+---------------------------- [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides] --------------------------------------------------------
+
+
+CREATE TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides](
```

Comment:
Here again, I think we should have a unique constraint on `ClaimSetResourceClaimActionAuthorization_ClaimSetResourceClaimActionAuthorizationId` and `AuthorizationStrategy_AuthorizationStrategyId`. It should not be possible to add the same authorization strategy twice here for a particular `ClaimSetResourceClaimActionAuthorizationId`.

@vimayya Thoughts?